flow_judge/models/huggingface.py

```diff
@@ -1,43 +1,122 @@
 import logging
 import os
-from typing import Any
+from typing import Any, Dict, Optional
+
+from pydantic import BaseModel
+
+try:
+    import torch
+    from huggingface_hub import snapshot_download
+    from transformers import AutoModelForCausalLM, AutoTokenizer
+
+    HF_AVAILABLE = True
+except ImportError:
+    HF_AVAILABLE = False
 
-import torch
-from huggingface_hub import snapshot_download
 from tqdm import tqdm
-from transformers import AutoModelForCausalLM, AutoTokenizer
 
-from flow_judge.models.base import BaseFlowJudgeModel
+from flow_judge.models.common import BaseFlowJudgeModel, ModelConfig, ModelType
 
 logger = logging.getLogger(__name__)
 
 
-class FlowJudgeHFModel(BaseFlowJudgeModel):
+class GenerationParams(BaseModel):
+    temperature: float = 0.1
+    top_p: float = 0.95
+    max_new_tokens: int = 1000
+    do_sample: bool = True
+
+
+class HfConfig(ModelConfig):
+    def __init__(
+        self,
+        model_id: str,
+        generation_params: GenerationParams,
+        device_map: str = "auto",
+        torch_dtype: str = "bfloat16",
+        flash_attn: bool = True,
+        **kwargs: Any,
+    ):
+        super().__init__(model_id, ModelType.TRANSFORMERS, generation_params.model_dump(), **kwargs)
+        self.device_map = device_map
+        self.torch_dtype = torch_dtype
+        self.flash_attn = flash_attn
+        self.kwargs = kwargs
+
+
+class Hf(BaseFlowJudgeModel):
     """FlowJudge model class for Hugging Face Transformers."""
 
-    def __init__(self, model_id: str, generation_params: dict[str, Any], **hf_kwargs: Any):
+    def __init__(
+        self,
+        model_id: str = None,
+        generation_params: Dict[str, Any] = None,
+        flash_attn: bool = True,
+        **kwargs: Any,
+    ):
         """Initialize the FlowJudge Hugging Face Transformers model."""
-        super().__init__(model_id, "transformers", generation_params, **hf_kwargs)
+        if not HF_AVAILABLE:
+            raise HfError(
+                status_code=1,
+                message="The required Hugging Face packages are not installed. "
+                "Please install them by adding 'hf' to your extras:\n"
+                "pip install flow-judge[...,hf]",
```

Comment:
Okay this is probably misleading, I meant to add hf into the already existing extras

`pip install flow-judge[hf]` would be wrong as that could install the other extras, but I don't know which ones the user has active.

Any ideas?

Reply:
I was actually referring to pip install flow-judge[...,hf] instead of pip install -e ".[...,hf]"

But you raised a valid point as well...

Reply:
I don't think we can know which extras are installed right?

Reply:
Oh yes, makes sense. Planning though to have the package in Pypi too as we anyway build it for testing it out and having the badges, I think it would help environments like Colab, Deepnote etc to install quickly.

Reply:
I'll fix this

Reply:
Oh yes in that case we should go with flow-judge. I had discussed with M but he said we should wait until a more stable version of the codebase.

Reply:
changed extras to work with pip and not uv pip